### PR TITLE
✨Source Fastbill : Migrate Python CDK to Low-code CDK 

### LINKED ISSUE #29277: Fastbill
https://docs.airbyte.com/integrations/sources/fastbill

### PR DESCRIPTION
**Pull request by avirajsingh7**
## What
Migrating Source fastbill to Low-Code CDK
Closes #29277  

## How
Developed using (Configuration Based Source) low-code CDK

## Recommended reading order

1 manifest.yaml
2 schemas/*

## 🚨 User Impact 🚨

*Are there any breaking changes? What is the end result perceived by the user?*
Yes, there are breaking changes due to Changes to Stream Schema
At many places previous data type were wrong and added some new properties in schema as per response from API

 ##  Test
<details>
<summary>Acceptance Test</summary>

</details>
Screenshot 2023-08-13 at 2 08 45 AM

<details>
<summary>Migrate Connector : Source Fastbill</summary>

- [x] **Community Member? Grant Edit Access to Maintainers [instructions](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests)**
- [x] **Secrets in the Connector's Spec**
- [x] **Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run ./gradlew :airbyte-integrations:connectors:<name>:integrationTest.**
- [x] **Code Reviews Completed**
  - [x] **Documentation Updated**
  - [x] **Connector's README.md**
  - [x] **Connector's bootstrap.md**
  - [x] **docs/integrations/<source or destination>/<name>.md (including changelog. See changelog example)**
  - [x] **docs/integrations/README.md**
  - [x] **airbyte-integrations/builds.md**
- [x] **PR Name follows PR Naming Conventions**
</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Hello 👋 , right now Airbyte's integration tests are failing for this connector and I'm going to work during the week to fix them. Due a high number of contributions we have a backlog and your contribution will be review next week. We ask to you share the integration test from your side whatever possible and if you don't have credentials to run tests to get in touch with @RealChrisSean as soon as possible. Also some common missing steps contributors does migrating to low code are (and you can double check to make sure you implemented this): 
* adding paginator
* update the documentation changelog + metadata file and dockerfile version
* if the connector has `start_date` parameter you must implement incremental syncs

**Comment by avirajsingh7** [comment]
> ight now Airbyte's integration tests are failing for this connector and I'm going to work during the week to fix them. Due a high number of contributions we have a backlog and your contribution will be review next week. We ask to you share the integration test from your side whatever possible and if you don't have credentials to run tests to get in touch with @RealChrisSean as soon as possible. Also some common missing steps contributors does migrating to low code are (and you can double check to make sure you implemented this):
> 
> * adding paginator
> * update the documentation changelog + metadata file and dockerfile version
> * if the connector has `start_date` parameter you must implement increme

Hey @marcosmarxm thanks for confirmation, confused from some days why integration test are failing.
Do let us know how can we help...

File: airbyte-integrations/connectors/source-fastbill/source_fastbill/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    type: DeclarativeStream
+    retriever:
+      $ref: "#/definitions/retriever"
+
+  invoices_stream:
+    $ref: "#/definitions/base_stream"
+    name: "invoices"
+    primary_key: "INVOICE_ID"
+    $parameters:
+      path: "/api.php"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fastbill/source_fastbill/manifest.yaml:45]: Maybe move the path to the base url. All streams use the same path.

File: airbyte-integrations/connectors/source-fastbill/source_fastbill/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      field_path: ["RESPONSE","{{ parameters.record_extractor }}"]
+  requester:
+    type: HttpRequester
+    url_base: "https://my.fastbill.com/api/1.0"
+    http_method: "POST"
+    authenticator:
+      class_name: source_fastbill.components.CustomAuthenticator
+      username: "config['username']"
+      password: "config['api_key']"
+    request_body_json:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fastbill/source_fastbill/manifest.yaml:17]: Can you also add the request header application/json here?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
A few comments after we can proceed to merge the contribution

File: airbyte-integrations/connectors/source-fastbill/source_fastbill/manifest.yaml (hunk 1 above)
Comment (avirajsingh7) [reply airbyte-integrations/connectors/source-fastbill/source_fastbill/manifest.yaml:45]: @marcosmarxm At first I have done that only but path can't be empty that's why I have to add this in path

Comment (avirajsingh7) [reply airbyte-integrations/connectors/source-fastbill/source_fastbill/manifest.yaml:45]: `path: ""` KEEPING PATH LIKE THIS ALSO NOT WORKING

Comment (marcosmarxm) [reply airbyte-integrations/connectors/source-fastbill/source_fastbill/manifest.yaml:45]: I see, thanks! Let's keep that as is

File: airbyte-integrations/connectors/source-fastbill/source_fastbill/manifest.yaml (hunk 2 above)
Comment (avirajsingh7) [reply airbyte-integrations/connectors/source-fastbill/source_fastbill/manifest.yaml:17]: Done added

**Comment by avirajsingh7** [comment]
@marcosmarxm done changes

File: airbyte-integrations/connectors/source-fastbill/source_fastbill/components.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+from dataclasses import dataclass
+import base64 
+from airbyte_cdk.sources.declarative.auth.token import BasicHttpAuthenticator
+from airbyte_cdk.sources.declarative.types import Config
+from isodate import Duration, parse_duration
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fastbill/source_fastbill/components.py:5]: Remove imports not used.

File: airbyte-integrations/connectors/source-fastbill/source_fastbill/components.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+
+@dataclass
+class CustomAuthenticator(BasicHttpAuthenticator):
+
+    config: Config
+    @property
+    def token(self) -> str:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fastbill/source_fastbill/components.py:13]: Please add a unit test to validate the custom function.

File: airbyte-integrations/connectors/source-fastbill/setup.py [hunk 5]
```diff
@@ -6,10 +6,18 @@
 from setuptools import find_packages, setup
 
 MAIN_REQUIREMENTS = [
-    "airbyte-cdk~=0.4",
+    "airbyte-cdk~=0.1",
 ]
 
+<<<<<<< HEAD
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fastbill/setup.py:12]: Fix this

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-fastbill/source_fastbill/components.py (hunk 3 above)
Comment (avirajsingh7) [reply airbyte-integrations/connectors/source-fastbill/source_fastbill/components.py]: removed

File: airbyte-integrations/connectors/source-fastbill/setup.py (hunk 5 above)
Comment (avirajsingh7) [reply airbyte-integrations/connectors/source-fastbill/setup.py]: fixed

File: airbyte-integrations/connectors/source-fastbill/source_fastbill/components.py (hunk 4 above)
Comment (avirajsingh7) [reply airbyte-integrations/connectors/source-fastbill/source_fastbill/components.py:12]: @marcosmarxm added unit tests

File: airbyte-integrations/connectors/source-fastbill/source_fastbill/components.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+        username = self.config['username'].encode("latin1")
+        password = self.config['api_key'].encode("latin1")
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fastbill/source_fastbill/components.py:15]: Please check the `BasicHttpAuthenticator` implementation. You need to use interpolated strings here and in the function below.

File: airbyte-integrations/connectors/source-fastbill/source_fastbill/manifest.yaml [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+      type: DpathExtractor
+      field_path: ["RESPONSE","{{ parameters.record_extractor }}"]
+  requester:
+    type: HttpRequester
+    url_base: "https://my.fastbill.com/api/1.0"
+    http_method: "POST"
+    authenticator:
+      class_name: source_fastbill.components.CustomAuthenticator
+      username: "config['username']"
+      password: "config['api_key']"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fastbill/source_fastbill/manifest.yaml:16]: This need to be a interpolated string `"{{ config['username'] }}"`

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments.

File: airbyte-integrations/connectors/source-fastbill/source_fastbill/components.py (hunk 6 above)
Comment (avirajsingh7) [reply airbyte-integrations/connectors/source-fastbill/source_fastbill/components.py]: @marcosmarxm changed

**Comment by marcosmarxm** [comment]
Merged in https://github.com/airbytehq/airbyte/pull/29767 thanks for the contribution @avirajsingh7

**Pull request closed without merging**